7 configuration

start974 opened the pull request:
Close #7

start974 committed:
update readme configuration

start974 committed:
update README with configuration table

start974 committed:
change format of table

start974 committed:
change configuration table and check key types

start974 committed:
Add config a file

start974 committed:
remove in README configuration table a line

remove line only used in vscoq2 (vscode)
and update lsp part

start974 commented:
Hi @tomtomjhj,

Could you check if all configuration keys can be used in vscoq.nvim and add a warning for any features that are not yet supported? I'm available to make modifications if needed!

start974 committed:
fix error on continuous and manual and add type

add type to meta file

start974 commented:
Ok I introduce a bug on continuous mod I try to fix it

start974 committed:
fix continuous mod not move in panel

tomtomjhj committed:
review and add TODOs

tomtomjhj commented:
Thanks! I added a commit with some TODO comments. can you take a look?

> Could you check if all configuration keys can be used in vscoq.nvim and add a warning for any features that are not yet supported? 

Good idea. I'll take look soon


note to myself: this is breaking change since the config uses the enumItemLabels instead of the index

start974 committed:
change config

make a real object for Config and LspOptions
use config in client

start974 committed:
update README

start974 commented:
Thanks for your review!

I have updated `Config` and `LspConfig` (now `lspOptions`) to be proper objects and ensured that `Config` is passed to the client.
I also updated the README based on your feedback.

Let me know if any further changes are needed!

tomtomjhj committed:
clean

tomtomjhj commented:
Thanks! 

I just notice that vscoq's readme has descriptions for the configuration. So why don't we simply link to that instead of repeating it here? But of course we'd need to list the differences (the `lsp` stuff and unsupported stuff).

> Could you check if all configuration keys can be used in vscoq.nvim and add a warning for any features that are not yet supported? I'm available to make modifications if needed!

Here are unsupported options. Can you add warnings for them?
* `vscoq.goals.display`
* `vscoq.proof.display-buttons`

start974 commented:
Yes, we can simply link to the existing *vscoq README* and list the differences. However, I don't think this is the optimal approach, as having the configuration listed directly avoids requiring users to switch pages while reading. Additionally, it will allow us to add new configuration keys in the future as we introduce new functionalities.

Regarding the unsupported options:
- `vscoq.proof.display-buttons` seems specific to *vscode* and could be removed.
- `vscoq.goals.display` could potentially be implemented in the future using a "Fold" or "Unfold" mechanism to collapse or expand other goals, but for now, it can be removed.

What do you think about it?

tomtomjhj commented:
> Yes, we can simply link to the existing vscoq README and list the differences. However, I don't think this is the optimal approach, as having the configuration listed directly avoids requiring users to switch pages while reading. Additionally, it will allow us to add new configuration keys in the future as we introduce new functionalities.

Makes sense!

> Regarding the unsupported options:

Sounds good

start974 committed:
remove goals display keys

remove key goals.displays
fix typos

start974 committed:
update lsp options

update following protocol:
https://github.com/coq/vscoq/blob/main/language-server/protocol/settings.ml

start974 commented:
I have removed the `goals.display` keys from the README and configuration.

I also removed some keys from `LspOptions` based on the [vscoq language server settings](https://github.com/coq/vscoq/blob/main/language-server/protocol/settings.ml).
However, I think the [protocol documentation](https://github.com/coq/vscoq/blob/main/docs/protocol.md) is incomplete. :/
The removed keys are:
- `goals.diff.maxDepth` I think we do not use this key also
- `proof.cursor.sticky`

tomtomjhj commented:
Some configs are not in vscoq server because they are only relevant to the client. `proof.cursor.sticky` one of such options, and used in vscoq.nvim.

start974 commented:
Yes, sure! That’s why I removed them from `LspOptions` but not from `Configuration`.

Feel free to comment on any other changes in this PR, I can update them as needed. 
I'm also motivated to help you implement new features! 🚀
I'll also be creating new issues soon.

tomtomjhj commented:
Merged. Thanks!
